test/LetConst/funcDeclConflict.js

```diff
@@ -0,0 +1,149 @@
+
+// Top level function in script does not conflict with same-named var
+WScript.LoadScript("function f2_v(){};");
+
+// Top level function in eval does not conflict with another top level function
+eval("function f3_f(){};");
+
+// Top level function in eval does not conflict with same-named var
+eval("function f4_v(){};");
+
+// Top level function in script conflicts with top level let
+try { WScript.LoadScript("function f5_l(){};"); } catch (e) { print(e); }
+
+// Top level function in script conflicts with top level const
+try { WScript.LoadScript("function f6_c(){};"); } catch (e) { print(e); }
+
+// Top level function in eval conflicts with top level let
+try { eval("function f5_l(){};") } catch (e) { print(e); }
+
+// Top level function in eval conflicts with top level const
+try { eval("function f6_c(){};") } catch (e) { print(e); }
+
+(function ff() {
+    if (true) {
+        let fo5_l = 123;
+
+        if (true) {
+            // TODO: this is blocked by https://github.com/Microsoft/ChakraCore/issues/5070
+            //
+            // Top level function in eval conflicts with outer function level let
+            try { eval("function fo5_l(){};") } catch (e) { print(e); }
+        }
+    }
+
+    if (true) {
+        // Top level function in eval conflicts with outer function level const
+        try { eval("function fo6_c(){};") } catch (e) { print(e); }
+    }
+
+    const fo6_c = 123;
+})();
+
+(function ffs() {
+    'use strict';
+
+    let fos5_l = 123;
+
+    // Top level function in eval conflicts with outer function level let (strict)
+    eval("function fos5_l(){};");
+
+    if (true) {
+        // Top level function in eval conflicts with outer function level const (strict)
+        eval("function fos6_c(){};");
+    }
+
+    const fos6_c = 123;
+})();
+
+
+(function ffl() {
+    let fo5_l_sl = 123;
+
+    // Top level function in eval conflicts with outer function level let
+    try { WScript.LoadScript("function fo5_l_sl(){};") } catch (e) { print(e); }
+
+    // Top level function in eval conflicts with outer function level const
+    try { WScript.LoadScript("function fo6_c_sl(){};") } catch (e) { print(e); }
+
+    const fo6_c_sl = 123;
+})();
+
+(function ffsl() {
+    'use strict';
+
+    let fos5_l = 123;
+
+    // Top level function in eval conflicts with outer function level let (strict)
+    WScript.LoadScript("function fos5_l_sl(){};");
+
+    // Top level function in eval conflicts with outer function level const (strict)
+    WScript.LoadScript("function fos6_c_sl(){};");
+
+    const fos6_c_sl = 123;
+})();
+
+(function ffn() {
+    let fo5_l_nf = 123;
+
+    // Top level function in "new Function" does not conflict with outer function level let
+    f = (new Function("return function fo5_l_nf(){};"))();
+
+    // Top level function in "new Function" does not conflict with outer function level const
+    f = (new Function("return function fo6_c_nf(){};"))();
+
+    console.log(f);

```

Comment:
>console.log [](start = 4, length = 11)

Nitpick : no need to print. You can do the f.toString() with the expected function string. And the end you can print('Pass') that you don't have to generate baseline.